### ISSUE TITLE
S2S connection between two ejabberd's fails

### ISSUE THREAD

**Issue by wornet-mwo** [processone/ejabberd#2219]
The problem is based on connection between two hosts.

> What version of ejabberd are you using?

17.12 in both setups

> What operating system (version) are you using?

alpine, both setups based on the docker image ejabberd/ecs:17:12, running on ubuntu-docker-host

> How did you install ejabberd (source, package, distribution)?

ejabberd/ecs:17:12 from hub.docker.com

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

S2S connections fail from both sides. The receiving side logs this:

```
2018-01-12 09:11:28.728 [error] <0.1725.0> gen_server <0.1725.0> terminated with reason: {'module could not be loaded',[{regexp,sh_to_awk,["*"],[]},{ejabberd_regexp,sh_to_awk,1,[{file,"src/ejabberd_regexp.erl"},{line,91}]},{xmpp_stream_pkix,match_labels,2,[{file,"src/xmpp_stream_pkix.erl"},{line,235}]},{lists,any,2,[{file,"lists.erl"},{line,1225}]},{xmpp_stream_pkix,authenticate,2,[{file,"src/xmpp_stream_pkix.erl"},{line,52}]},{xmpp_stream_in,process_sasl_request,2,[{file,"src/xmpp_stream_in.erl"},{line,788}]},{xmpp_stream_in,handle_info,2,[{file,"src/xmpp_stream_in.erl"},{line,...}]},...]}
2018-01-12 09:11:28.729 [error] <0.1725.0>@regexp:sh_to_awk CRASH REPORT Process <0.1725.0> with 0 neighbours exited with reason: call to undefined function regexp:sh_to_awk("*") in p1_server:terminate/7 line 878
2018-01-12 09:11:28.730 [error] <0.1541.0>@regexp:sh_to_awk Supervisor ejabberd_s2s_in_sup had child undefined started with {ejabberd_s2s_in,start_link,undefined} at <0.1725.0> exit with reason call to undefined function regexp:sh_to_awk("*") in context child_terminated
```

The connecting side:

```
2018-01-12 09:08:53.227 [info] <0.1733.0>@ejabberd_s2s_out:init:277 Outbound s2s connection started: CCC.de -> YYY.net
2018-01-12 09:08:53.273 [info] <0.1733.0>@ejabberd_s2s_out:process_closed:161 Failed to establish outbound s2s connection CCC.de -> YYY.net: Stream reset by peer; bouncing for 89 seconds
```

Connections from other domains servers (non ejabberd) are working.

Maybe related settings:

´´´
s2s_use_starttls: optional
´´´

Additionally on CCC.de:
```
acl:
  good_s2s_peers:
    server:
      - "YYY.net"
      - "jabber.YYY.net"
acccess:
  s2s:
    good_s2s_peers: allow
    all: deny
```

I'll provide additional config etc. anonymized on request (as there is a customers system involved) and this is quite hard to do in advance. Thanks in advance for help.

**Comment by zinid** [comment]
Clearly installation problem: some bytecode files are missing.

**Comment by cromain** [comment]
mix.exs lacks xmerl application

**Comment by cromain** [comment]
xmerl is used only to call xmerl_regexp:sh_to_awk/1. maybe this can be rewritten to avoid need to include all xmerl by the way.

**Comment by cromain** [comment]
fixed by 3df78d3a8. We removed xmerl dependency.
